tests/unit/test_parallel_write.py

```diff
@@ -0,0 +1,185 @@
+        io.write(dynamic_table, number_of_jobs=number_of_jobs)
+
+    # TODO: roundtrip currently fails due to read error
+    #with ZarrIO(path=zarr_top_level_path, mode="r") as io:
+    #    dynamic_table_roundtrip = io.read()
+    #    data_roundtrip = dynamic_table_roundtrip["TestColumn"].data
+    #    assert_array_equal(data_roundtrip, data)
+
+
+@unittest.skipIf(not TQDM_INSTALLED, "optional tqdm module is not installed")
+def test_simple_tqdm(tmpdir):
+    number_of_jobs = 2
+    expected_desc = f"Writing Zarr datasets with {number_of_jobs} jobs"
+
+    zarr_top_level_path = str(tmpdir / "test_simple_tqdm.zarr")
+    with patch("sys.stderr", new=StringIO()) as tqdm_out, ZarrIO(path=zarr_top_level_path,  manager=get_manager(), mode="w") as io:
+        column = VectorData(
+            name="TestColumn",
+            description="",
+            data=PickleableDataChunkIterator(
+                data=np.array([1., 2., 3.]),
+                display_progress=True,
+                #progress_bar_options=dict(file=tqdm_out),
+            )
+        )
+        dynamic_table = DynamicTable(name="TestTable", description="", columns=[column])
+        io.write(dynamic_table, number_of_jobs=number_of_jobs)
+
+    assert expected_desc in tqdm_out.getvalue()
+
+
+@unittest.skipIf(not TQDM_INSTALLED, "optional tqdm module is not installed")
+def test_compound_tqdm(tmpdir):
+    number_of_jobs = 2
+    expected_desc_pickleable = f"Writing Zarr datasets with {number_of_jobs} jobs"
+    expected_desc_not_pickleable = "Writing non-parallel dataset..."
+
+    zarr_top_level_path = str(tmpdir / "test_compound_tqdm.zarr")
+    with patch("sys.stderr", new=StringIO()) as tqdm_out, ZarrIO(path=zarr_top_level_path,  manager=get_manager(), mode="w") as io:
+        pickleable_column = VectorData(
+            name="TestGenericColumn",
+            description="",
+            data=PickleableDataChunkIterator(
+                data=np.array([1., 2., 3.]),
+                display_progress=True,
+            )
+        )
+        not_pickleable_column = VectorData(
+            name="TestClassicColumn",
+            description="",
+            data=NotPickleableDataChunkIterator(
+                data=np.array([4., 5., 6.]),
+                display_progress=True,
+                progress_bar_options=dict(desc=expected_desc_not_pickleable, position=1)
+            )
+        )
+        dynamic_table = DynamicTable(name="TestTable", description="", columns=[pickleable_column, not_pickleable_column])
+        io.write(dynamic_table, number_of_jobs=number_of_jobs)
+
+    tqdm_out_value = tqdm_out.getvalue()
+    assert expected_desc_pickleable in tqdm_out_value
+    assert expected_desc_not_pickleable in tqdm_out_value
+
+
+def test_extra_args(tmpdir):
+    pass # TODO? Should we test if the other arguments like thread count can be passed?
+    # I mean, anything _can_ be passed due to dynamic **kwargs, but how to test if it was actually used? Seems difficult...
```

Comment:
@oruebel thoughts on if tests for this are needed, or strategy to take?

For the `max_threads_per_process` and `multiprocessing_context` arguments that are usually encouraged to just default, but control is left to allow them to be overridden for architectures that utilize it

Reply:
Yes, testing those particular parameters is tricky when they only being passed in function calls during `ZarrIO.write` where you don't have an easy way to inspect those values. The question is, does `max_threads_per_process` need to be set separately for every call to `exhaust_queue` (i.e., does this change between calls) or should this really be a property of the `ZarrIODataChunkIteratorQueue` instead? If the latter, then you could check after you call `ZarrIO.write` whether `ZarrIO.__dci_queue.max_threads_per_process` has the correct setting. 

Otherwise, I would add empty tests for the cases that you would like to check and skip those checks and then add a comment why this is not being tested.

Reply:
It seems as if the `ZarrIODataChunkIteratorQueue` [is only instantiated at time](https://github.com/hdmf-dev/hdmf-zarr/blob/dev/src/hdmf_zarr/backend.py#L118) of `ZarrIO` instantiation; whereas both practically and in terms of design, each of these keyword arguments (the most important of which, `number_of_jobs`) are specific to call to the method `io.write` (which then deploys to `write_group`/`write_builder`, etc.) as opposed to being properties of the Queue or by extension, the IO class itself

This is also to avoid confusion regarding other IO operations, such as `io.read` - if we had `ZarrIO(number_of_jobs=4, ...)` one might think any operation performed using that IO would be parallelized when in reality it would still only be the `write` call

I suppose I could still set them as instance attributes of the `ZarrIODataChunkIteratorQueue` class that are always `None` until the call to `exhaust_queue` with `number_of_jobs > 1` + whatever extra argument specifications, and then update the attributes to non-`None` values in that event, which would at least let us test their propagation to the right level

How does that sound?

Reply:
> whereas both practically and in terms of abstract design, each of these keyword arguments (the most important of which, `number_of_jobs`) are specific to the call to the method `io.write`

That makes sense. To take this line-of-thought another step further, I think `ZarrIODataChunkIteratorQueue`  as a whole is really specific to `io.write`. As such, an alternative approach could be to instead of in `ZarrIO.__init__` we could create a new `ZarrIODataChunkIteratorQueue` every time we call `io.write` and allow the user to optionally supply their own `ZarrIODataChunkIteratorQueue`. I.e., something like `io.write(dci_queue=ZarrIODataChunkIteratorQueue(number_of_jobs=4)`.  In this way, the user would supply settings like `number_of_jobs` (which are specific to `ZarrIODataChunkIteratorQueue`) explicitly to the queue, rather than indirectly via `io.write`.  What do you think?

Reply:
I think that makes more sense; I wasn't entirely sure if it would work so I tried it safely on a separate branch, moving this PR over to that

#118